chatlas/_chat.py

```diff
@@ -823,6 +832,176 @@ async def extract_data_async(
+        )
+
+        response = await session.list_tools()
+        for tool in response.tools:
+            if include_tools:
+                if tool.name not in include_tools:
+                    continue
+            if exclude_tools:
+                if tool.name in exclude_tools:
+                    continue
+            self._tools[tool.name] = Tool.from_mcp(
+                session=session,
+                mcp_tool=tool,
+            )
+
+    async def register_sse_mcp_server_async(
+        self,
+        name: str,
+        url: str,
+        include_tools: Optional[list[str]] = None,
+        exclude_tools: Optional[list[str]] = None,
+        transport_kwargs: dict[str, Any] = {},
+    ):
+        """
+        Register a SSE-based MCP server session asynchronously.
+
+        This method establishes a new SSE (Server-Sent Events) connection to an MCP server and registers
+        the available tools. The server is identified by a unique name and URL.
+
+        Parameters
+        ----------
+        name
+            Unique identifier for this MCP server session
+        url
+            URL endpoint of the MCP server
+        include_tools
+            List of tool names to include. If None, all available tools will be included. Defaults to None.
+        exclude_tools
+            List of tool names to exclude. This parameter and include_tools are mutually exclusive. Defaults to None.
+        transport_kwargs
+            Additional keyword arguments to pass to the SSE transport layer. Defaults to {}.
+
+        Raises
+        ------
+        AssertionError
+            If a session with the given name already exists
+
+        Returns
+        -------
+        None
+
+        Examples
+        --------
+        ```python
+        await chat.register_sse_mcp_server_async(
+            name="my_server",
+            url="http://localhost:8080/sse",
+            include_tools=["tool1", "tool2"],
+            transport_kwargs={"timeout": 30},
+        )
+        ```
+        """
+        assert name not in self._mcp_sessions, f"Session {name} already exists."
+
+        transport = await self._mcp_exit_stack.enter_async_context(
+            sse_client(url, **transport_kwargs)
+        )
+        self._mcp_sessions[name] = await self._mcp_exit_stack.enter_async_context(
+            MCPClientSession(*transport)
+        )
+        session = self._mcp_sessions[name]
+        await session.initialize()
+
+        await self._register_mcp_tools(
+            session,
+            include_tools=include_tools,
+            exclude_tools=exclude_tools,
+        )
+
+    async def register_stdio_mcp_server_async(
+        self,
+        name: str,
+        command: str,
+        args: list[str],
+        env: dict[str, str] | None = None,
+        include_tools: Optional[list[str]] = None,
+        exclude_tools: Optional[list[str]] = None,
+        transport_kwargs: dict[str, Any] = {},
```

Comment:
Is there any way we can support typing on these arguments?


```suggestion
        kwargs: dict[str, Any] = {},
```